x/stakeibc/keeper/hooks_deposit_records_test.go

```diff
@@ -0,0 +1,527 @@
+	priorEpoch := currentEpoch - 1
+	emptyDepositRecords := []recordstypes.DepositRecord{
+		{
+			Id:                 1,
+			Amount:             0,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+		{
+			Id:                 2,
+			Amount:             0,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+	}
+
+	recordsToBeTransfered := []recordstypes.DepositRecord{
+		{
+			Id:                 3,
+			Amount:             3000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+		{
+			Id:                 4,
+			Amount:             4000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+	}
+	transferAmount := sdk.NewInt(3000 + 4000)
+
+	recordsToBeStaked := []recordstypes.DepositRecord{
+		{
+			Id:                 5,
+			Amount:             5000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: priorEpoch,
+		},
+		{
+			Id:                 6,
+			Amount:             6000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: priorEpoch,
+		},
+	}
+	stakeAmount := sdk.NewInt(5000 + 6000)
+
+	recordsInCurrentEpoch := []recordstypes.DepositRecord{
+		{
+			Id:                 7,
+			Amount:             7000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: currentEpoch,
+		},
+		{
+			Id:                 8,
+			Amount:             8000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: currentEpoch,
+		},
+	}
+
+	return TestDepositRecords{
+		emptyRecords:          emptyDepositRecords,
+		recordsToBeTransfered: recordsToBeTransfered,
+		recordsToBeStaked:     recordsToBeStaked,
+		recordsInCurrentEpoch: recordsInCurrentEpoch,
+		transferAmount:        transferAmount,
+		stakeAmount:           stakeAmount,
+	}
+}
+
+func (s *KeeperTestSuite) SetupDepositRecords() DepositRecordsTestCase {
+	delegationAccountOwner := fmt.Sprintf("%s.%s", HostChainId, "DELEGATION")
+	delegationChannelID := s.CreateICAChannel(delegationAccountOwner)
+	delegationAddress := s.IcaAddresses[delegationAccountOwner]
+
+	ibcDenomTrace := s.GetIBCDenom(Atom) // we need a true IBC denom here
+	hostModuleAddress := stakeibctypes.NewZoneAddress(HostChainId)
```

Comment:
nit: thoughts on `hostZoneModuleAddress` or `zoneModuleAddress`? I initially thought this account was on the host chain

Reply:
happy to rename it! Do you generally use zone when discussing something host-zone-related, but living on Stride? I had kind of though zone and hostZone were synonymous

Reply:
I think we do use them interchangeably, it would might be best to use host zone because that's the name of the message (so implies the data lives on stride)